### ISSUE TITLE
SonarQube deployment on OpenShift cluster fails

### ISSUE THREAD

**Issue by georgfedermann** [opendevstack/ods-core#562]
**Describe the bug**
Following the ODS installation instructions as provided here: https://www.opendevstack.org/ods-documentation/opendevstack/3.x/administration/installation.html.

After having successfully setup the ODS namespace and having installed nexus in OpenShift by running 
```
make install-ods-project 
make install-nexus 
make configure-nexus
```
the attempt to install SonarQube by running
```
make install-sonarqube
```
will fail in the target 
```
apply-sonarqube-deploy
```
after the SonarQube build has finished successfully. The deployment will hang with the single log line:
```
--> Scaling sonarqube-1 to 1
```

**To Reproduce**
1. In a fresh CentOS 7 box run the ods-core ods-devenv setup script to get an OpenShift cluster with ODS integration including a nexus deployment
2. In ods-core run ```make install-sonarqube``` 

**Expected behavior**
SonarQube will be deployed in a pod and become available as a service under the configured route.

**Affected version (please complete the following information):**
 - OpenShift: 3.11
 - OpenDevStack master

**Log Output (ensure to remove any confidential information like tokens, project names, etc.**
```
--> Scaling sonarqube-1 to 1
```

**Comment by clemensutschig** [comment]
Can you Check the deployment  Events

**Comment by georgfedermann** [comment]
These are the deployment events for sonarqube (1 event):
10:40:49 PM 	Normal 	Successful Create  	Created pod: sonarqube-1-5x86t
The deployment ends in a Crash Loop Back-off.

Meanwhile, the sonarqube-postgresql deployment finishes successfully:
10:40:48 PM 	Normal 	Successful Create  	Created pod: sonarqube-postgresql-1-x9s5f

I found these additional log entries when looking at the logs just before the pod got cleaned up:
```
2020.05.17 20:47:33 INFO  es[][o.e.b.BootstrapChecks] explicitly enforcing bootstrap checks
ERROR: [1] bootstrap checks failed
[1]: max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
```
This could be fixed by running this command which I've added to the setup script:
```
**sudo sysctl -w vm.max_map_count=262144**
```
Which makes the way free for the next error:
```
2020.05.17 21:23:31 ERROR web[][o.s.s.p.PlatformImpl] Web server startup failed
java.lang.IllegalStateException: Fail to connect to database
Caused by: java.lang.IllegalStateException: Can not connect to database. Please check connectivity and settings (see the properties prefixed by 'sonar.jdbc.').
Caused by: java.sql.SQLException: Cannot create PoolableConnectionFactory (FATAL: password authentication failed for user "sonarqube")
Caused by: org.postgresql.util.PSQLException: FATAL: password authentication failed for user "sonarqube"
```
Assuming the credentials are taken from ods-core.env to configure the postgresql user and the SonarQube jdbc client:
```
SONAR_DATABASE_PASSWORD_B64=c29uYXJxdWJlCg==
SONAR_DATABASE_USER=sonarqube
```
Verifying the credentials by manually logging into the db in the pod shows that no password is set for the sonarqube user. After manually setting the password to the value configured in ods-core.env the SonarQube deployment passes jdbc client setup. **I would ask for the postgres user credentials configuration to be fixed. Should I create a new issue for that?** In that case, we could close this issue.

Now, the SonarQube setup fails with:
```
2020.05.17 21:50:16 ERROR web[][o.s.s.p.Platform] Background initialization failed. Stopping SonarQube
org.sonar.api.utils.SonarException: Security realm fails to start: Unable to test connection to crowd
```
So, we need to setup Crowd first before proceeding with SonarQube setup.

**Comment by michaelsauter** [comment]
The `vm.max_map_count` issue is fixed in OpenShift 4 with the `tuning` label. In OpenShift 4, that label has no effect so your node needs to have this prerequisite. I will note that in the installation instructions.

The Postgres user should have a password set, see https://github.com/opendevstack/ods-core/blob/master/sonarqube/ocp-config/sonarqube.yml#L145-L149. Are you really sure about this?

And yes, if `https://github.com/opendevstack/ods-core/blob/master/configuration-sample/ods-core.env.sample#L85` is set to `true`, you'll need Crowd to be there.

**Comment by georgfedermann** [comment]
@michaelsauter I will double-check the sonarqube password issue again right away and keep you posted. Yet, that's what the installer has been running into.

Thanks for pointing out the config item in https://github.com/opendevstack/ods-core/blob/master/configuration-sample/ods-core.env.sample#L85! This should unblock the SQ setup task for now.

**Comment by michaelsauter** [comment]
Closing as the requirement has been added to the install guide.
